src/Module/Contact/Profile.php

```diff
@@ -136,6 +136,10 @@ protected function post(array $request = [])
 			Contact\User::setChannelFrequency($cdata['user'], $this->session->getLocalUserId(), $_POST['channel_frequency']);
 		}
 
+		if (isset($_POST['channel_only'])) {
+			Contact\User::setChannelOnly($cdata['user'], $this->session->getLocalUserId(), $_POST['channel_only']);
```

Comment:
`$_POST` is available as `$request` in this method`.
```suggestion
		if (isset($request['channel_only'])) {
			Contact\User::setChannelOnly($cdata['user'], $this->session->getLocalUserId(), $request['channel_only']);
```

Reply:
Done